src/lib/y2partitioner/clients/main.rb

```diff
@@ -19,57 +19,97 @@
+      # run only if the user accepts the warning. The storage stack needs to be
+      # initialized in read-write access mode.
+      #
+      # @return [Boolean]
+      def run_partitioner?
+        start_partitioner_warning == :yes &&
+          setup_storage_manager
+      end
 
-        return nil unless Y2Storage::StorageManager.setup(mode: :rw)
+      # Tries to initialize the storage stack
+      #
+      # @return [Boolean] true if storage was initialized in rw mode;
+      #   false otherwise.
+      def setup_storage_manager
+        Y2Storage::StorageManager.setup(mode: :rw)
+      end
+
+      # @return [Y2Storage::StorageManager]
+      def storage_manager
+        Y2Storage::StorageManager.instance
+      end
 
-        smanager = Y2Storage::StorageManager.instance
-        dialog = Dialogs::Main.new
-        res = dialog.run(smanager.probed, smanager.staging)
+      # Saves on disk all changes performed by the user
+      def commit
+        storage_manager.staging = partitioner_dialog.device_graph
+        storage_manager.commit
+      end
 
-        # Running system: presenting "Expert Partitioner: Summary" step now
-        # ep-main.rb SummaryDialog
-        if res == :next && should_commit?(allow_commit)
-          smanager.staging = dialog.device_graph
-          smanager.commit
-        end
+      # Partitioner dialog is initalized with the probed and staging devicegraphs
+      #
+      # @return [Dialogs::Main]
+      def partitioner_dialog
+        Dialogs::Main.new(storage_manager.probed, storage_manager.staging)
       end
 
-      # Ask whether to proceed with changing the disks;
-      # or inform that we will not do it.
-      # @return [Boolean] proceed
-      def self.should_commit?(allow_commit)
-        if allow_commit
-          q = "Modify the disks and potentially destroy your data?"
-          Yast::Popup.ContinueCancel(q)
-        else
-          m = "Nothing gets written, because the device graph is fake."
-          Yast::Popup.Message(m)
-          false
-        end
+      # Popup to alert the user about using the Partitioner
+      #
+      # @return [Symbol] user's answer (:yes, :no)
+      def start_partitioner_warning
+        message = _(
+          "Only use this program if you are familiar with partitioning hard disks.\n\n" \
+          "Never partition disks that may, in any way, be in use\n" \
+          "(mounted, swap, etc.) unless you know exactly what you are\n" \
+          "doing. Otherwise, the partitioning table will not be forwarded to the\n" \
+          "kernel, which will most likely lead to data loss.\n\n" \
+          "To continue despite this warning, click Yes."
+        )
+
+        Yast2::Popup.show(message, headline: :warning, buttons: :yes_no)
+      end
+
+      # Popup when commit is forbidden (e.g., when the client is used for manual testing)
+      def forbidden_commit_warning
+        message = _("Nothing gets written because commit is not allowed.")
```

Comment:
NP: maybe a comment for translators (also useful for developers) so they are away this is not really a string to be displayed to normal users, just something for the test mode.